pkg/utils/map.go

```diff
@@ -56,14 +58,18 @@ func cleanUpMapValue(v interface{}) interface{} {
 }
 
 func Dedupe(l []string) []string {
-	res := make([]string, 0)
-	seen := make(map[string]bool)
-	for _, val := range l {
-		if _, ok := seen[val]; ok {
-			continue
-		}
-		res = append(res, val)
-		seen[val] = true
+	return sets.NewString(l...).List()
+}
+
+type SimpleType interface {
+	string | int
+}
+
+func Map[T any, R SimpleType](slice []T, mapper func(elem T) R) []R {
```

Comment:
We should make sure goreleaser is configured for the latest golang version to handle this

Reply:
I'm pretty sure it is already.

Reply:
I see that `go.mod` was also updated to use Go `1.18` already, that's why I wanted to try the new generics thingy :slightly_smiling_face:

Reply:
Oh awesome